### PR TITLE
Git Actions

### PR DESCRIPTION
**Pull request by ConradWeiser**
# Github Actions Artifact Publications
When a push is made to the Github sources repository, if it is pushed associated with a tag matching v* (eg. v0.0.1, v2.0, etc.) the system will perform the following:
* Run all of the unit tests associated in the project
* Run the Gulp buildfile, bundling all of the available sources
* Create a release on the Github repository with all of the artifacts uploaded

Note that the publication will not occur if any of the unit tests fail. Eg.
Image

## Accessing the data once published
As each of the releases are published with their own specific version number, there is a magic link to retrieve just the latest. However you need to postprocess the data a little in order to parse out the download URLs.

If you make a GET request to the repository at address (adjusted to your repository): `https://api.github.com/repos/conradweiser/extensions/release/latest`, you retrieve back a Json object. ([Pastebin Link as it's a little large](https://pastebin.com/j2nFD2b4))

You may parse the response relatively easily to retrieve what you're searching for. Insides of the assets JsonArray, you are able to located the name of the file (which matches gulpfile output exactly), as well as it's browser_download_url, this is a URL which can be used to download the file contents directly.

## Note
If you really must, you may remove the unit testing line from the .yml file, as currently it does fail. However unless you're using it for development purposes, I would not recommend it. I've gone in manually and verified the errors that it's returning, which turn out to be valid. 
Bad sources likely would cause issues clientside.

If you would like to see how the releases page looks when published, see [this link here](https://github.com/ConradWeiser/extensions/releases) - Note that only passing builds are labeled latest, and populated with artifacts. Definitely run your unit tests locally and verify that your stuff works before uploading.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+on:
+  push:
+    tags:
+    - 'v*'
```
Comment (DanielKovalevich) [inline .github/workflows/main.yml:4]: We want the action to execute whenever a pull request has been accepted and merged

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    steps:
+      - name: Checkout Branch
+        uses: actions/checkout@v2
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v1.4.2
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - run: npm install
+      - run: npm test
```
Comment (DanielKovalevich) [inline .github/workflows/main.yml:25]: The tests will take too long in our case. Normally, this would not be an issue but we will need to temporarily disable this functionality.

File: .github/workflows/main.yml (hunk 1 above)
Comment (ConradWeiser) [reply .github/workflows/main.yml:4]: In order for releases to be published along with artifacts, it is a __requirement__ that you have a tag associated. After a lot of experimentation, I've discovered that there's no way to get around this, unfortunately. 

One of my experiments was to create a tag called 'release', and to simply hardcode each build release, as something under that tag. It worked once. It didn't work a second time, not allowing an overwrite. 

The bottom line, there is no one step solution to make everything run appropriately on pull request. You can get close by manually assigning tags through a git client of some sort BEFORE passing the request, although at that point, why use the merge trigger in general. You'd have more fine control over the release process if you stick with just tag triggers.

**Comment by ConradWeiser** [comment]
Branch is temporarily dirty, do not merge

**Comment by ConradWeiser** [comment]
The release generation Workflow has been swapped out with [a Git Commit action](https://github.com/stefanzweifel/git-auto-commit-action), this will build, and publish the latest artifacts in your repositories `bundles/` directory.

This runs on push, and on git merge. 

The links should be static to access.

File: .github/workflows/main.yml (hunk 1 above)
Comment (ConradWeiser) [reply .github/workflows/main.yml]: Tagging system has been replaced with on push or pull request resolution
